### PR TITLE
Human-friendly labels for all nodes

### PR DESCRIPTION
**Pull request by cmc333333**
Don't display "1111_222-V-B-3", display "Section V.B.3".

To get there, split HTMLBuilder into some subclasses.

Resolves eregs/notice-and-comment#49

### CODE REVIEW

File: regulations/generator/html_builder.py [hunk 1]
```diff
@@ -128,6 +96,64 @@ def process_node(self, node):
...
+            self.process_node(c)
+
+    @staticmethod
+    def human_label(node):
+        """Derive a human-readable description for this node"""
+        return '-'.join(node['label'])      # Default
+
+
+class CFRHTMLBuilder(HTMLBuilder):
+    header_regex = re.compile(r'^(§&nbsp;)(\s*\d+\.\d+)(.*)$')
```
Comment (jmcarp) [inline regulations/generator/html_builder.py]: Where is this used?

```bash
± ag -G py header_regex
htmlcov/regulations_generator_html_builder_py.html
255:<p id="t15" class="stm run hide_run">&nbsp; &nbsp; <span class="nam">header_regex</span> <span class="op">=</span> <span class="nam">re</span><span class="op">.</span><span class="nam">compile</span><span class="op">(</span><span class="str">ur&#39;^(&#167;&amp;nbsp;)(\s*\d+\.\d+)(.*)$&#39;</span><span class="op">)</span><span class="strut">&nbsp;</span></p>

regulations/generator/html_builder.py
19:    header_regex = re.compile(r'^(§&nbsp;)(\s*\d+\.\d+)(.*)$')
```

File: regulations/generator/html_builder.py [hunk 2]
```diff
@@ -125,6 +95,59 @@ def process_node(self, node):
...
+            'part': self.tree['label'][0],
+            'reg_name': ''
+        }
+        reg_title = self.parse_doc_title(self.tree['title'])
+        if reg_title:
+            titles['reg_name'] = reg_title
+        return titles
+
+    def parse_doc_title(self, reg_title):
+        match = re.search(r"[(].+[)]$", reg_title)
```
Comment (jmcarp) [inline regulations/generator/html_builder.py]: Nit-picking: why use character classes for parentheses here? Could we describe the same rule with the slightly simpler and arguably more idiomatic `\(.+\)$`? Also, I get that this just getting moved around and not new.

**Comment by jmcarp** [comment]
LGTM--left a very minor comment about some old code.

File: regulations/generator/html_builder.py (hunk 2 above)
Comment (cmc333333) [inline regulations/generator/html_builder.py]: Good point. Updated!

**Comment by jmcarp** [comment]
Thanks for humoring me @cmc333333!

**Pull request merged**